lib/spack/spack/build_environment.py

```diff
@@ -733,6 +734,11 @@ def setup_package(pkg, dirty):
 
         load_external_modules(pkg)
 
+    implicit_rpaths = pkg.compiler.implicit_rpaths()
```

Comment:
What is changed by moving this block from `set_build_environment_variables` to `setup_package`? I assume nothing since this method calls that method. Is it just clearer to place it here?

Reply:
It now happens after the modules for the compiler have been loaded.